### PR TITLE
[FIX] mrp: bom properties

### PR DESCRIPTION
**Pull request by simongoffin**
The bom related to the product in a sale line order must be
filtered by the function _bom_find.

opw:632558

### CODE REVIEW

**Comment by simongoffin** [comment]
@beledouxdenis

File: addons/mrp/stock.py [hunk 1]
```diff
@@ -70,12 +55,15 @@ def _action_explode(self, cr, uid, move, context=None):
         uom_obj = self.pool.get("product.uom")
         to_explode_again_ids = []
         processed_ids = []
-        bis = self._check_phantom_bom(cr, uid, move, context=context)
+        property_ids = map(int, move.procurement_id.sale_line_id.property_ids or [])
```
Comment (Gorash) [inline addons/mrp/stock.py]: property_ids inside the if (no request if no bom)

**Comment by Gorash** [comment]
Ok (better if you add a test)

**Comment by beledouxdenis** [comment]
I would appreciate @qdp-odoo & @csnauwaert opinion on this one, to make sure we are not missing a use case here, since they are the author of this whole `_check_phantom_bom` thing.

Indeed, from the code, it seems clearly wanted for the `bis` variable to be of type `phantom`, trhough the domain tuple `('type', '=', 'phantom')`.

This is no longer the case with this PR, and I am afraid we are missing a functional concept here.

**Comment by csnauwaert** [comment]
If I remember correctly, it is important to keep the type=phantom

Typically, let's assume you are selling "desktop chairs" and that desktop chair has a BOM telling you that you need 5 wheels, the chairs, etc. You have 2 use cases here:
- If the BOM is set to type=normal, you sell the final product so you build the chair and sell it to the customer. So you want to see on your picking the product named "chair"
- If the BOM is a phantom bom, you sell the pieces to your customer and they build it themselve (like IKEA), in this case you want to see on your picking not the chair product but the pieces details (5 wheels, and so on..).

So when validating a picking, if you have some product with a BOM, you only "explode" the one with a phantom BOM.

@qdp-odoo will correct me if I'm wrong

**Comment by Gorash** [comment]
@csnauwaert Yes, but i's a check after the bom_find, because if you have more of one bom for a product you need to select the bom who match and then explode or not in function of it's a phantom or not.

=> Need to use SUPERUSER_ID for _bom_find
=> add a if bom_point.type == "phantom" after the find

File: addons/mrp/stock.py [hunk 2]
```diff
@@ -70,12 +55,14 @@ def _action_explode(self, cr, uid, move, context=None):
...
-        if bis:
-            bom_point = bom_obj.browse(cr, SUPERUSER_ID, bis[0], context=context)
+        property_ids = map(int, move.procurement_id.sale_line_id.property_ids or [])
+        bis = bom_obj._bom_find(cr, uid, product_id=move.product_id.id, properties=property_ids)
+        bom_point = bom_obj.browse(cr, SUPERUSER_ID, bis, context=context)
+        if bis and bom_point.type=='phantom':
             factor = uom_obj._compute_qty(cr, SUPERUSER_ID, move.product_uom.id, move.product_uom_qty, bom_point.product_uom.id) / bom_point.product_qty
-            res = bom_obj._bom_explode(cr, SUPERUSER_ID, bom_point, move.product_id, factor, [], context=context)
-            
+            property_ids = map(int, move.procurement_id.sale_line_id.property_ids or [])
```
Comment (Gorash) [inline addons/mrp/stock.py]: why again ?

**Pull request merged**